bootstrap.js

```diff
@@ -0,0 +1,46 @@
+/* global Components */
+
+const { classes: Cc, interfaces: Ci } = Components;
+const DefaultUA = Cc["@mozilla.org/network/protocol;1?name=http"]
+  .getService(Ci.nsIHttpProtocolHandler).userAgent;
+const ObserverService = Cc["@mozilla.org/observer-service;1"].getService(Ci.nsIObserverService);
+
+class UAOverrider {
+  init() {
+    ObserverService.addObserver(this, "http-on-useragent-request", false);
+  }
+
+  uninit() {
+    ObserverService.removeObserver(this, "http-on-useragent-request");
+  }
+
+  observe(subject, topic) {
+    if (topic != "http-on-useragent-request") {
+      return;
+    }
+
+    let channel = subject.QueryInterface(Ci.nsIHttpChannel),
+      uaOverride = this.getUAForURI(channel.URI);
+
+    if (uaOverride) {
+      channel.setRequestHeader("User-Agent", uaOverride, false);
+    }
+  }
+
+  getUAForURI() {
+    return "GoFaster Test - " + DefaultUA;
```

Comment:
Not too sure yet. Having a large list of URIs is one option, but if we do want to support globbing, we'd have to be a little bit more flexible.

At the moment, I'm thinking about a per-override validation function that receives the URL and can return a boolean which can trigger further processing. This may be a bit more complicated in the beginning, but having a URI->UA mapping table would be only an interim solution since we do have other requirements (like patching CSS or JS) as well. I'll investigate some ways to keep the performance impact low and start another discussion on Bugzilla when I'm ready.